app/api/server/v1/voip/omnichannel.ts

```diff
@@ -0,0 +1,134 @@
+		}));
+		const { username } = this.requestParams();
+		const user = await Users.findOneByAgentUsername(username, {
+			projection: { _id: 1 },
+		});
+		if (!user) {
+			return API.v1.notFound('User not found');
+		}
+		const extension = await Users.getVoipExtensionByUserId(user._id, {
+			projection: {
+				_id: 1,
+				username: 1,
+				extension: 1,
+			},
+		}).toArray();
+		if (!extension) {
+			return API.v1.notFound('Extension not found');
+		}
+		return API.v1.success({ extension });
+	},
+
+	// Create agent-extension association.
+	async post() {
+		if (!hasPermission(this.userId, 'manage-agent-extension-association')) {
+			return API.v1.unauthorized();
+		}
+		check(this.bodyParams, {
+			username: String,
+			extension: String,
+		});
+		const { username, extension } = this.bodyParams;
+		const user = await Users.findOneByAgentUsername(username, {
+			projection: {
+				_id: 1,
+				username: 1,
+			},
+		});
+		if (!user) {
+			return API.v1.notFound();
+		}
+		await Users.setExtension(user._id, extension);
+		return API.v1.success();
+	},
+	async delete() {
+		if (!hasPermission(this.userId, 'manage-agent-extension-association')) {
+			return API.v1.unauthorized();
+		}
+		check(this.requestParams(), Match.ObjectIncluding({
+			username: String,
+		}));
+		const { username } = this.requestParams();
+		const user = await Users.findOneByAgentUsername(username, {
+			projection: {
+				_id: 1,
+				username: 1,
+			},
+		});
+		if (!user) {
+			return API.v1.notFound();
+		}
+		await Users.unsetExtension(user._id);
+		return API.v1.success();
+	},
+});
+
+// Get free extensions
+API.v1.addRoute('omnichannel/extension', { authRequired: true,
+	permissionsRequired: ['manage-agent-extension-association'],
+}, {
+	async get() {
+		check(this.queryParams, Match.ObjectIncluding({
+			type: String,
+			agentId: Match.Maybe(String),
+		}));
+		const { type, username } = this.queryParams;
+		switch ((type as string).toLowerCase()) {
+			case 'free': {
+				const extension = await LivechatVoip.getFreeExtensions();
+				if (!extension) {
+					return API.v1.failure('Error in finding free extensons');
+				}
+				return API.v1.success({ extensions: extension.result });
+			}
+			case 'allocated': {
+				const association = await LivechatVoip.getExtensionAllocationDetails();
+				if (!association) {
+					return API.v1.failure('Error in allocated extensions');
+				}
+				return API.v1.success({ allocations: association.result });
+			}
+			case 'available': {
+				const user = await Users.findOneByAgentUsername(username, {
+					projection: { _id: 1 },
+				});
+				if (!user) {
+					return API.v1.notFound('User not found');
+				}
+				const extension = await Users.getVoipExtensionByUserId(user._id, {
+					projection: {
+						_id: 1,
+						username: 1,
+						extension: 1,
+					},
+				}).toArray();
+				const freeExt = await LivechatVoip.getFreeExtensions();
+				const extensions = [];
+				for (let i = 0; i < extension.length; i++) {
```

Comment:
I'd say you can benefit from the spread syntax here 👀 For example:

```
const available = [...freeExt, ...extension];
return API.v1.success({ available });
```

Reply:
Correct me if I am wrong, but that may not work here by just using spread syntax. Extension is an allocated extension and it is coming from the database. freeExt is coming from connector and it is string array. So spread syntax my not just work.